coalib/bears/Bear.py

```diff
@@ -274,6 +281,284 @@ def log_message(self, log_message, timestamp=None, **kwargs):
+                    os.chdir(subdir)
+                filename = '{}_{}.prof'.format(
+                    kwargs['section_name'], kwargs['bear_name'])
+                open(filename, 'w+')
+                kwargs.pop('section_name')
+                kwargs.pop('bear_name')
+                cProfile.runctx('self.run(*args, **kwargs)',
+                                globals(), locals(), filename)
+                os.chdir("..")
+            return False, args, kwargs
+        if str(profile_output) is not u'False':
+            str_profile_bears = str(profile_bears[0]).lower().strip()
+            prof = cProfile.Profile()
+            kwargs.pop('section_name')
+            kwargs.pop('bear_name')
+            kwargs.pop('profile_dump')
+            kwargs.pop('profile_commands')
+            if not str_profile_bears == u'false':
+                if (u'wrapping_function of' in str(self.run)) or (u'_new_func of' in str(self.run)):
+                    kwargs['profiler'] = prof
+                else:
+                    kwargs.pop('profile_output')
+                    prof.enable()
+                retval = self.run(*args, **kwargs)
+                if inspect.isgenerator(retval):
+                    retval, retval_clone = itertools.tee(retval)
+                    for i in retval_clone:
+                        pass
+                prof.disable()
+                check_trim = True if 'no_trim' in profile_bears else False
+                if str_profile_bears == u'true':
+                    self.setup_profiler_table(prof, profile_bears, check_trim)
+                else:
+                    try:
+                        with open(str_profile_bears, 'a') as stream:
+                            ps = self.pstats_config(prof, stream, profile_bears)
+                            if check_trim:
+                                ps.print_stats()
+                            else:
+                                ps.print_stats()
+                            stream.close()
+                    except FileNotFoundError:
+                        self.err('No such file or directory: "{}", '
+                                 'the first argument to `--profile-bears`'
+                                 ' must be true or a valid file path'.format(
+                                     str_profile_bears))
+                return retval, args, kwargs
+            else:
+                kwargs.pop('profile_output')
+                return False, args, kwargs
+        if str(profile_commands) is not u'False':
+            print(profile_bears)
+            str_profile_bears = str(profile_bears[0]).lower().strip()
+            prof = cProfile.Profile()
+            kwargs.pop('section_name')
+            kwargs.pop('bear_name')
+            kwargs.pop('profile_dump')
+            kwargs.pop('profile_output')
+            if str_profile_bears is not None:
+                if (u'wrapping_function of' in str(self.run)) or (u'_new_func of' in str(self.run)):
```

Comment:
Line is longer than allowed. (100 > 80)

*Origin: LineLengthBear, Section: `linelength`.*